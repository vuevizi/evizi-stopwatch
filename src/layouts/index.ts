import MainLayout from "./MainLayout/MainLayout";
import Header from "./Header/Header";
import Footer from "./Footer/Footer";
export {Header,MainLayout,Footer}