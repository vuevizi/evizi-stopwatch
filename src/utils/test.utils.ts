export const findTestElement = (component: any, attr: string) => {
  const findElement: any = component.find(`[data-test="${attr}"]`);
  return findElement;
};
