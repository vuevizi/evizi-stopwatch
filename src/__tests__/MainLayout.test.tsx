import {MainLayout,Header,Footer} from "../layouts";
import {findTestElement} from "../utils/test.utils";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
const setUpComponent = (props={}) => {
    const component:any = shallow(<MainLayout {...props}/>)
    return component;
}

describe("Main Layout",()=>{
    let component:any;
    beforeEach(()=>{
        component = setUpComponent();
    });
    test("renders without crash",()=>{
        expect(component.length).toBe(1);
    })
    test("renders the Header correctly",()=>{
        const header = shallow(<Header/>);
        expect(header.length).toBe(1);
    });
    test("renders the Footer correctly",()=>{
        const footer = shallow(<Footer/>);
        expect(footer.length).toBe(1);
    })
    test("renders the Main Content correctly",()=>{
        const mainContent = findTestElement(component,"main-content");
        expect((mainContent.length)).toBe(1);
    })
    test("renders Watch and Table Timer in Main Content",()=>{
        const watch = findTestElement(component,"watch");
        const tableTimer = findTestElement(component,"table-timer");
        expect(watch.length).toBe(1);
        expect(tableTimer.length).toBe(1);
    })
    test("renders match to snapshots",()=>{
        expect(toJson(component)).toMatchSnapshot();
    })
})
