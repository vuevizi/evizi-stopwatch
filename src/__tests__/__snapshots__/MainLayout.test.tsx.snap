// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout renders match to snapshots 1`] = `
<Fragment>
  <Header
    data-test="header"
  />
  <styled.section
    data-test="main-content"
  >
    <Watch
      data-test="watch"
    />
    <TableTimer
      data-test="table-timer"
    />
  </styled.section>
  <Footer
    data-test="footer"
  />
</Fragment>
`;

exports[`Main Layout renders without crash 1`] = `
<Fragment>
  <Header />
  <styled.section>
    <Watch />
    <TableTimer />
  </styled.section>
  <Footer />
</Fragment>
`;
